### DISCUSSION TITLE
blogs or articles on how linfree is made, it's system design, how the UI is designed and the whole Infrastructure.

### DISCUSSION THREAD

**Discussion by GauravBurande** [EddieHubCommunity/LinkFree#4288] [Ideas]
I started contributing to open source just 2 days ago.

I decided to start contributing to linkfree, I figured out some of the codebase, but still, there are some things that I don't understand which are new to me, and this happens with so many new contributors.

So, I have an idea that the maintainers and the builders of this webapp should write blogs or articles and explain how linkfree is built.
That way if any newcomer has any problems understanding the linkfree codebase, instead of the newcomer searching docs or blogs related to the tech he/she didn't understand somewhere on the internet or google, they can just read the blog or articles. So, there won't be any barrier which causes new contributors to lose their motivation or confidence and delay their first open-source contribution.

This will cause more and more developers to contribute to open source and I think this should be applied to every open-source project.

Please share your thoughts on this, your feedback will be really appreciated.

**Comment by eddiejaoude** [comment]
Hi, thank you for the suggestion. I have created videos and live streams on it, and we also have official docs. Can you be specific about what is missing?

Here are the docs https://linkfree.eddiehub.io/docs

**Comment by GauravBurande** [comment]
I haven't seen your videos but I've read the docs.
I didn't quite understand the backend code. 
The docs provide information about testing and playing with the frontend. But still it doesn't describe the logic behind building this webapp.

Like how single user mode is implemented. What's the logger and the logic of backend.

For example this blog:
https://gauravv.hashnode.dev/heres-how-i-built-swagstoar-webapp

You may have explained all of this already in your videos, but I haven't watched those videos yet, I'll check it out.

**Reply by eddiejaoude** [reply]
You are correct, I haven't gone into that sort of details yet, thank you for the suggestion 👍

**Reply by GauravBurande** [reply]
Glad that I could help.
